engine/src/main/java/com/redhat/cloud/notifications/db/repositories/EndpointRepository.java

```diff
@@ -45,106 +40,99 @@
      * multiple endpoints and recipients settings. The properties created below have no impact on the resolution of the
      * action recipients.
      */
-    public Uni<Endpoint> getOrCreateDefaultEmailSubscription(String accountId) {
+    public Endpoint getOrCreateDefaultEmailSubscription(String accountId) {
         String query = "FROM Endpoint WHERE accountId = :accountId AND compositeType.type = :endpointType";
-        return commonStateSessionFactory.withSession(true, session -> {
-            return session.createQuery(query, Endpoint.class)
-                    .setParameter("accountId", accountId)
-                    .setParameter("endpointType", EMAIL_SUBSCRIPTION)
-                    .getResultList()
-                    .call(this::loadProperties)
-                    .onItem().transformToUni(emailEndpoints -> {
-                        EmailSubscriptionProperties properties = new EmailSubscriptionProperties();
-                        Optional<Endpoint> endpointOptional = emailEndpoints
-                                .stream()
-                                .filter(endpoint -> properties.hasSameProperties(endpoint.getProperties(EmailSubscriptionProperties.class)))
-                                .findFirst();
-                        if (endpointOptional.isPresent()) {
-                            return Uni.createFrom().item(endpointOptional.get());
-                        }
-
-                        Endpoint endpoint = new Endpoint();
-                        endpoint.setProperties(properties);
-                        endpoint.setAccountId(accountId);
-                        endpoint.setEnabled(true);
-                        endpoint.setDescription("System email endpoint");
-                        endpoint.setName("Email endpoint");
-                        endpoint.setType(EMAIL_SUBSCRIPTION);
-                        properties.setEndpoint(endpoint);
-
-                        return session.persist(endpoint)
-                                .call(() -> session.persist(endpoint.getProperties()))
-                                .replaceWith(endpoint);
-                    });
-        });
+        List<Endpoint> emailEndpoints = statelessSessionFactory.getCurrentSession().createQuery(query, Endpoint.class)
+                .setParameter("accountId", accountId)
+                .setParameter("endpointType", EMAIL_SUBSCRIPTION)
+                .getResultList();
+        loadProperties(emailEndpoints);
+
+        EmailSubscriptionProperties properties = new EmailSubscriptionProperties();
+        Optional<Endpoint> endpointOptional = emailEndpoints
+                .stream()
+                .filter(endpoint -> properties.hasSameProperties(endpoint.getProperties(EmailSubscriptionProperties.class)))
+                .findFirst();
+        if (endpointOptional.isPresent()) {
+            return endpointOptional.get();
+        }
+
+        Endpoint endpoint = new Endpoint();
```

Comment:
we could put this endpoint creation in a separate method.

Reply:
This PR is only focused on dropping the reactiveness. I'm not changing old code unless it has a direct impact on the reactiveness drop.